README.md

```diff
@@ -404,73 +404,71 @@ Here is an example of headers needed to sign a request.
-  * 'keyId' is a unique id, formatted as an url. hostname is used to retrieve the public key via custom discovery
-  * 'algorithm' specify the algorithm used to generate signature
-  * 'headers' specify the properties used when generating the signature
-  * 'signature' the signature of an array containing the properties listed in 'headers'. Some properties like content-length, date, digest, and host are mandatory to protect against authenticity override.
-
+- `(request-target)` contains the reached endpoint and the used method,
+- `content-length` is the total length of the payload of the request,
+- `date` is the date and time when the request has been sent,
+- `digest` is a checksum of the payload of the request,
+- `host` is the hostname of the recipient of the request (remote when signing outgoing request, local on incoming request),
+- `Signature` contains the signature generated using the private key and details on its generation:
+  * `keyId` is a unique id, formatted as an URL; hostname is used to retrieve the public key via custom discovery
+  * `algorithm` specify the algorithm used to generate signature
+  * `headers` specify the properties used when generating the signature
+  * `signature` the signature of an array containing the properties listed in `headers`. Some properties like content-length, date, digest, and host are mandatory to protect against authenticity override.
 
 ### How to generate the Signature for outgoing request
 
 After properties are set in the headers, the Signature is generated and added to the list.
 
-This is a quick PHP example of headers for outgoing request:
-
-```php
-    $headers = [
-        '(request-target)' => 'post /path',
-        'content-length' => strlen($payload),
-        'date' => gmdate('D, d M Y H:i:s T'),
-        'digest': 'SHA-256=' . base64_encode(hash('sha256', utf8_encode($payload), true)),
-        'host': 'hostname.of.the.recipient',
-    ];
-
-    openssl_sign(implode("\n", $headers), $signed, $privateKey, OPENSSL_ALGO_SHA256);
-
-    $signature = [
-        'keyId' => 'https://author.hostname/key',
-        'algorithm' => 'rsa-sha256',
-        'headers' => 'content-length date digest host',
-        'signature' => $signed
-    ];
-
-    $headers['Signature'] = implode(',', $signature);
+This is a pseudo-code example for generating the `Signature` header for outgoing requests:
+
+```code
+headers = {
+    '(request-target)': 'post /path',
+    'content-length': length_of(payload),
+    'date': current_gmt_datetime(),  # Use a function to get the current GMT date as 'D, d M Y H:i:s T'
+    'digest': 'SHA-256=' + base64_encode(hash('sha256', utf8_encode(payload))),
+    'host': 'recipient-hostname',
+}
+
+signed = ssl_sign(concatenate_with_newlines(headers), private_key, 'sha256')
+signature = {
+    'keyId': 'https://sender.org/ocm',  # The sending server's keyId as advertised at the discovery endpoint
```

Comment:
Reviewing myself, I modified this to clarify that the keyId is per sending server (and not per user)